analyzers/src/SonarAnalyzer.CSharp/Rules/CollectionEmptinessChecking.cs

```diff
@@ -36,81 +36,45 @@ public sealed class CollectionEmptinessChecking : SonarDiagnosticAnalyzer
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
-                {
-                    var binary = (BinaryExpressionSyntax)c.Node;
-                    CheckCountOne(binary.Left, binary.Right, c);
-                    CheckCountZero(binary.Right, binary.Left, c);
-                },
-                SyntaxKind.LessThanOrEqualExpression);
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
-                {
-                    var binary = (BinaryExpressionSyntax)c.Node;
-                    CheckCountZero(binary.Left, binary.Right, c);
-                    CheckCountZero(binary.Right, binary.Left, c);
-                },
-                SyntaxKind.EqualsExpression);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
                     var binary = (BinaryExpressionSyntax)c.Node;
-                    CheckCountZero(binary.Left, binary.Right, c);
-                    CheckCountZero(binary.Right, binary.Left, c);
+
+                    if (ExpressionNumericConverter.TryGetConstantIntValue(binary.Left, out var left))
+                    {
+                        CheckExpression(c, binary.Right, left, CSharpFacade.Instance.Syntax.ComparisonKind(binary).Mirror());
+                    }
+                    else if (ExpressionNumericConverter.TryGetConstantIntValue(binary.Right, out var right))
+                    {
+                        CheckExpression(c, binary.Left, right, CSharpFacade.Instance.Syntax.ComparisonKind(binary));
+                    }
                 },
+                SyntaxKind.GreaterThanExpression,
+                SyntaxKind.GreaterThanOrEqualExpression,
+                SyntaxKind.LessThanExpression,
+                SyntaxKind.LessThanOrEqualExpression,
+                SyntaxKind.EqualsExpression,
                 SyntaxKind.NotEqualsExpression);
-        }
 
-        private static void CheckCountZero(ExpressionSyntax zero, ExpressionSyntax count, SyntaxNodeAnalysisContext context)
-        {
-            if (ExpressionNumericConverter.TryGetConstantIntValue(zero, out var value) &&
-                value == 0 &&
-                TryGetCountCall(count, context.SemanticModel, out var reportLocation, out var typeArgument))
-            {
-                context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, reportLocation, typeArgument));
-            }
-        }
-        private static void CheckCountOne(ExpressionSyntax one, ExpressionSyntax count, SyntaxNodeAnalysisContext context)
+        private static void CheckExpression(
+            SyntaxNodeAnalysisContext context,
+            ExpressionSyntax expression,
+            int constant,
+            ComparisonKind comparison)
```

Comment:
Should be a single line, it doesn't violate S103
```suggestion
        private static void CheckExpression(SyntaxNodeAnalysisContext context, ExpressionSyntax expression, int constant, ComparisonKind comparison)
```